docs/Compiler-Hardening-Guides/Compiler-Annotations-for-C-and-C++.md

```diff
@@ -0,0 +1,42 @@
+
+Table 1: Recommended attributes
+
+| Compiler Flag                                     |       Supported by    | Type     | Description                                                                         |
+|:------------------------------------------------- |:---------------------:|:--------:|:----------------------------------------------------------------------------------- |
+| `malloc`                                          | GCC<br/>Clang 13 | Function | Allocator that returns a non-aliased (possibly NULL) pointer.                   |
+| `malloc (dealloc)`                                | GCC 11 | Function | Same as above but also associate `dealloc` as the deallocator for this function |
+| `alloc_size(pos)`<br/>`alloc_size(pos-1, pos-2)`  | GCC <br/>Clang | Function | Size of the object that the returned pointer points to is at argument `pos` of the function or product of arguments at `pos-1` and `pos-2`. |
+| `access(mode, ref-pos)`<br/>`access(mode, ref-pos, size-pos)`  | GCC 10 | Function | Indicate how the function uses argument at `ref-pos`.  `mode` could be `read_only`, `read_write`, `write_only` or `none`.  `size-pos`, if mentioned, is the argument indicating the size of object at `ref-pos`. |
+| `fd_arg(N)`                                          | GCC 13 | Function | Argument N is an open file descriptor.                   |
+| `fd_arg_read(N)`                                          | GCC 13 | Function | Argument N is an open file descriptor that can be read from.                   |
+| `fd_arg_write(N)`                                          | GCC 13 | Function | Argument N is an open file descriptor that can be written to.                   |
+| `noreturn`                                          | GCC<br/>Clang | Function | The function does not return.                   |
+| `tainted_args`                                          | GCC 12<br/>Clang TBD | Function or function pointer | Function needs sanitization of its arguments. Used by -fanalyzer=taint                  |
+
+## Performance considerations
+
+Attributes influence not only diagnostics generated by the compiler but also the resultant code. As a result, annotating code with attributes will have an impact on performance, although the impact may go either way.  The annotation may allow compilers to add more traps for additional security and be conservative about some optimizations, thus impacting performance of output code. At the same time however, it may allow compilers to make some aggressive optimization decisions, resulting in generation of smaller and faster running code and often, better code layout.
+
+[^1]: Attribute Syntax
+<https://gcc.gnu.org/onlinedocs/gcc/Attribute-Syntax.html>
```

Comment:
Nit-pick: I started by placing both lines in one to make sure that Jekyll renders the footnote correctly - and while on it, I added some additional text to make the footnote consistent with other references to the GCC manual.

```suggestion
[^1]: GCC team, [Attribute Syntax](https://gcc.gnu.org/onlinedocs/gcc/Attribute-Syntax.html), GCC Manual, 2023-11-03.
```

Reply:
Thanks, fixed.